HACKING.md

```diff
@@ -0,0 +1,42 @@
+## Introduction
+LU is build with the [Entity–component–system](https://en.wikipedia.org/wiki/Entity_component_system) which you may know from the Unity Engine. In short that means everything that you see on screen (and more) is a GameObject on which Components can be attached. These components give the object functionality like rendering, physics and so on. 
+
+LU uses four different types of servers to communicate. These are: The authentication server (Uchu.Auth), which handles login and redirection to the character server (Uchu.Char). The character server then handles the character selection and redirection to the world server (Uchu.World). Now the world server is the server that you actually play on. It handles everything from movement to chat to missions. There can be multiple World servers running at once but one world server only serves one world. There can however be multiple world servers serving the same world. This happens if the maximum amount of players on a server is reached.
```

Comment:
as far as the client is aware there are 3 types of servers. Authentication, World (Character and World) and Chat (Never used/obsolete)

Reply:
Chat is handled by the world server right? And what about the UGC Server? Zaop told me about that one

Reply:
Chat is handled by the world server technically, chat packets are sent in a redirect packet that the world server is supposed to hand to the master server to handle

Reply:
and the UGC server isn't an actual RakNet server like the rest, it is just a webserver

Reply:
I updated the intro, what do you think?

Reply:
Seems good